src/languages/javascript.js

```diff
@@ -279,6 +279,6 @@ export default function(hljs) {
 
       }
     ],
-    illegal: /#(?!!)/
+    illegal: /#(?![!$_A-z])/
```

Comment:
We should be able to remove the ! now since we deal with shebang separately now.  `!` isn't allowed in normal identified afaik... can it be used with private fields somehow?

Reply:
> it's going to miss a lot... and also $...

Well I meant "in general". :)  Adding $ as you did is great.

> The actual spec allows a full range of Unicode characters ... So #$α is totally a legitimate name for a private class field. And in a project I'm working on, we actually do use Greek letters in certain spots. 😬 Not sure how to tackle that with a regex that handles it via negative lookahead.

Well you'd just add it to the range, no?  But we don't support the new `/u` modifier stuff yet... so you're stick with hard coding like Java does:

```
var JAVA_IDENT_RE = '[\u00C0-\u02B8a-zA-Z_$][\u00C0-\u02B8a-zA-Z_$0-9]*';
```

I'm find with tackling that matter separately though.

Reply:
Could we just remove the `!` here and then add an entry to CHANGES.md and then I think this could be merged.  Thanks.

Reply:
Yep, I’ll knock that out this evening or (more likely) tomorrow. Thanks!

Reply:
This'll need to be rebased now/rejiggered (since I just merged the TS/JS merge)... but hopefully won't be hard since the change to JS is only a one liner. :-)